tf2/include/tf2/time.h

```diff
@@ -52,8 +54,32 @@ namespace tf2
     return std::chrono::system_clock::now();
   }
 
-  inline double durationToSec(const tf2::Duration & input){
-    return (double)std::chrono::duration_cast<std::chrono::seconds>(input).count();
+  inline Duration durationFromSec(double t_sec)
+  {
+    int32_t sec, nsec;
+    sec = (int32_t)floor(t_sec);
+    nsec = (int32_t)std::nearbyint((t_sec-sec) * 1e9);
```

Comment:
> I also do not think cpplint.py checks this (it may not be possible to do so easily).

IIRC cpplint does check this, but I don't think we run linter on this package